private/bufpkg/bufconfig/buf_policy_yaml_file.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2020-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufconfig
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"slices"
+
+	"github.com/bufbuild/buf/private/pkg/encoding"
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"github.com/bufbuild/buf/private/pkg/syserror"
+)
+
+const (
+	defaultBufPolicyYAMLFileName    = "buf.policy.yaml"
+	defaultBufPolicyYAMLFileVersion = FileVersionV2
+)
+
+var (
+	bufPolicyYAMLFileNames                       = []string{defaultBufPolicyYAMLFileName}
+	bufPolicyYAMLFileNameToSupportedFileVersions = map[string]map[FileVersion]struct{}{
+		defaultBufPolicyYAMLFileName: {
+			FileVersionV2: struct{}{},
+		},
+	}
+)
+
+// BufPolicyYAMLFile represents a buf.policy.yaml file.
```

Comment:
Instead of `buf.policy.yaml`, could we say something like:

```
BufPolicyYAMLFile represents a policy configuration YAML file.
```

Just because this is meant to be generic.